adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/InternalsVisibleTo.cs

```diff
@@ -34,3 +34,4 @@
 [assembly: InternalsVisibleTo("Test.ADAL.NET.Unit, PublicKey=0024000004800000940000000602000000240000525341310004000001000100B5FC90E7027F67871E773A8FDE8938C81DD402BA65B9201D60593E96C492651E889CC13F1415EBB53FAC1131AE0BD333C5EE6021672D9718EA31A8AEBD0DA0072F25D87DBA6FC90FFD598ED4DA35E44C398C454307E8E33B8426143DAEC9F596836F97C8F74750E5975C64E2189F45DEF46B2A2B1247ADC3652BF5C308055DA9")]
 [assembly: InternalsVisibleTo("Test.ADAL.NET.Integration, PublicKey=0024000004800000940000000602000000240000525341310004000001000100B5FC90E7027F67871E773A8FDE8938C81DD402BA65B9201D60593E96C492651E889CC13F1415EBB53FAC1131AE0BD333C5EE6021672D9718EA31A8AEBD0DA0072F25D87DBA6FC90FFD598ED4DA35E44C398C454307E8E33B8426143DAEC9F596836F97C8F74750E5975C64E2189F45DEF46B2A2B1247ADC3652BF5C308055DA9")]
 [assembly: InternalsVisibleTo("WinFormsAutomationApp, PublicKey=0024000004800000940000000602000000240000525341310004000001000100B5FC90E7027F67871E773A8FDE8938C81DD402BA65B9201D60593E96C492651E889CC13F1415EBB53FAC1131AE0BD333C5EE6021672D9718EA31A8AEBD0DA0072F25D87DBA6FC90FFD598ED4DA35E44C398C454307E8E33B8426143DAEC9F596836F97C8F74750E5975C64E2189F45DEF46B2A2B1247ADC3652BF5C308055DA9")]
+[assembly: InternalsVisibleTo("Test.MSAL.NET.Integration, PublicKey=00240000048000009400000006020000002400005253413100040000010001002D96616729B54F6D013D71559A017F50AA4861487226C523959D1579B93F3FDF71C08B980FD3130062B03D3DE115C4B84E7AC46AEF5E192A40E7457D5F3A08F66CEAB71143807F2C3CB0DA5E23B38F0559769978406F6E5D30CEADD7985FC73A5A609A8B74A1DF0A29399074A003A226C943D480FEC96DBEC7106A87896539AD")]
```

Comment:
These are for the L1 tests, i.e. E2E headless. If you need internal relation, these are L0, and they should go to the unit test assembly.

Reply:
not sure what you mean here, please explain

Reply:
The basic idea is that integration tests should have no reason to have access to internals.

---
In reply to: [223127605](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r223127605) [](ancestors = 223127605)

Reply:
how would you check state of the cache or modify it (for example to expire AT to trigger RT usage)?

Reply:
You don't. Unit tests do that.

Reply:
This is integration testing of msal and adal coexistence  it is not unit test

Reply:
which is exactly why there should ideally not be a reference... but makes sense.

---
In reply to: [224193062](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r224193062) [](ancestors = 224193062)

Reply:
end to end test does not need access to internals, integration test can have access to it, discussed project called "Test.MSAL.NET.Integration"